stacks-signer/src/client/mod.rs

```diff
@@ -565,6 +565,7 @@ pub(crate) mod tests {
             tx_fee_ustx: config.tx_fee_ustx,
             max_tx_fee_ustx: config.max_tx_fee_ustx,
             db_path: config.db_path.clone(),
+            first_proposal_burn_block_timing: Duration::from_secs(30),
```

Comment:
For example, I think we should definitely try to avoid any situation where miner A mines block 0 of their tenure, then 10 seconds later mines block 1, then the bitcoin block arrives and both of these blocks are reorged. Like, this time should be less than 2 units of the expected block time.

We also don't want to make it so long that the miner can just wait until they see the next bitcoin block, then propose their first block. This would allow them to intentionally stall the network (by not creating blocks) and hurt the next miner (by causing their tenure to be invalid), without any risk to getting their block reward.

Reply:
On the other side, if we make this time too small, then the honest miner who submits their block commit quickly would be punished. Would this incentivize miners to wait whatever this timeout is after seeing a bitcoin block before they submit their block commit?

Reply:
Switched to checking when the proposal was signed off locally, which would let us use a tighter bound here I think (more like 20 seconds?) but I think ultimately picking a good default bound will be something that should happen during the testnet evaluations.